src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/Aes.cs

```diff
@@ -33,6 +35,470 @@ protected Aes()
+                return true;
+            }
+            catch
+            {
+                // It's only important to clear destination if it was not rented...
+                // but rather than have some exceptions clear it, and some not, always clear.`
+                CryptographicOperations.ZeroMemory(destination);
+                throw;
+            }
+            finally
+            {
+                lease.Dispose();
+            }
+        }
+
+        /// <summary>
+        ///   Unwraps a key that was wrapped using the IETF RFC 5649 AES Key Wrap with Padding algorithm.
+        /// </summary>
+        /// <param name="source">The data to unwrap.</param>
+        /// <param name="destination">
+        ///   The buffer to receive the unwrapped key.
+        ///   </param>
+        /// <returns>The number of bytes in the unwrapped key.</returns>
+        /// <exception cref="CryptographicException">
+        ///   <para>The unwrap algorithm failed to unwrap the ciphertext.</para>
+        ///   <para>-or-</para>
+        ///   <para>An error occurred during the cryptographic operation.</para>
+        /// </exception>
+        /// <remarks>
+        ///   <para>
+        ///     When called by the base class,
+        ///     <paramref name="source"/> is pre-validated to be at least 16 bytes long and a multiple of 8 bytes.
+        ///   </para>
+        ///   <para>
+        ///     When called by the base class,
+        ///     <paramref name="destination"/> will always be exactly 8 bytes shorter than <paramref name="source"/>,
+        ///     so any valid value will always fit.
+        ///   </para>
+        /// </remarks>
+        protected virtual int DecryptKeyWrapPaddedCore(ReadOnlySpan<byte> source, Span<byte> destination)
+        {
+            ulong iv;
+
+            if (source.Length == 16)
+            {
+                Span<byte> decrypt = stackalloc byte[16];
+                DecryptEcb(source, decrypt, PaddingMode.None);
+                iv = BinaryPrimitives.ReadUInt64BigEndian(decrypt);
+                decrypt.Slice(8).CopyTo(destination);
+            }
+            else
+            {
+                iv = Rfc3394Unwrap(source, destination);
+            }
+
+            uint len = (uint)iv;
+            ulong header = iv >> 32;
+            int slen = (int)len;
+
+            // Only 0..7 padding bytes are allowed.
+            // If len > maxOutput, that was "negative" padding, which is a large positive uint, so "more than 7".
+            // If len == maxOutput, pad is 0, which is valid.
+            // If len < maxOutput by less than 8, then pad is in the range 0..7, which is valid.
+            // If len is any lower than that, then pad is more than 7, which is invalid.
+            int maxOutput = source.Length - 8;
+            uint pad = (uint)maxOutput - len;
+
+            if (header != 0xA65959A6UL || pad > 7 || destination.Slice(slen).IndexOfAnyExcept((byte)0) >= 0)
```

Comment:
Should this do a MemoryMarshal to convert the unsigned longs to `ReadOnlySpan<byte>` and use `CryptographicOperations.FixedTimesEqual`? I don't think comparing two 64-bit integers is guaranteed to be a fixed time op on 32-bit platforms.

Reply:
(Or maybe some other architecture don't do it at all like certain RISC-V profiles or whatever). My earlier comment about bitness doesn't actually matter.

Reply:
My general intuition here is that since this basically serves as an authentication tag. See SP800-38F. Not constant time checking the "tag" feels off.

Reply:
OK, so you mean just on the A65959A6 part.  My reference to OSSL source is that they don't hide the difference between that being wrong, the padding count being out of bounds, or any padding bytes not being 0x00.

Switched the header comparison to use `uint`, so now it's a single instruction on 32-bit platforms as well as 64-bit.